tests/models/virtualfolder.py

```diff
@@ -152,3 +152,118 @@ def test_vfolder_with_no_filter_rules():
         vfolder.clean_fields()
 
     assert u'Some filtering rule must be specified.' in str(excinfo.value)
+
+
+@pytest.mark.django_db
+def test_vfolder_membership(site_matrix):
+    from pootle_pytest.factories import VirtualFolderFactory
+
+    from pootle_store.models import Unit
+    from pootle_store.util import OBSOLETE, TRANSLATED
+
+    vfolder = VirtualFolderFactory(filter_rules="store0.po")
+
+    live_units = Unit.objects.filter(state__gt=OBSOLETE)
+
+    expected_units = live_units.filter(store__name="store0.po")
+
+    # check default vfolder membership
+    assert (
+        sorted(vfolder.units.values_list("pk", flat=True))
+        == sorted(expected_units.values_list("pk", flat=True)))
+
+    vfolder.location = "/language0/{PROJ}/"
+    vfolder.save()
+
+    expected_units = live_units.filter(
+        store__translation_project__language__code="language0",
+        store__name="store0.po")
+
+    # check vfolder membership after changing the location
+    assert (
+        sorted(vfolder.units.values_list("pk", flat=True))
+        == sorted(expected_units.values_list("pk", flat=True)))
+
+    obsolete_unit = (
+        Unit.objects.filter(
+            state=OBSOLETE,
+            store__translation_project__language__code="language0",
+            store__name="store0.po"))[0]
+
+    # obsolete unit is not in the vfolder
+    assert obsolete_unit not in vfolder.units.all()
+
+    obsolete_unit.state = TRANSLATED
+    obsolete_unit.save()
+
+    # unobsoleted unit is in the vfolder
+    assert obsolete_unit in vfolder.units.all()
+
+    to_obsolete = vfolder.units.all()[0]
```

Comment:
no that is not a good idea - as currently there is an issue with mtime which is how first is defined.

this test really does not the obsolete unit was retrieved